backend/dataall/modules/shares_base/services/share_notification_service.py

```diff
@@ -35,7 +34,7 @@ class ShareNotificationService:
         - share.owner (person that opened the request) OR share.groupUri (if group_notifications=true)
     """
 
-    def __init__(self, session, dataset: S3Dataset, share: ShareObject):
+    def __init__(self, session, dataset, share: ShareObject):
```

Comment:
why remove the type?

Reply:
because it can be a dataset of a type that is not of S3 type. It should actually be a DatasetsBase type, but for that we need to modify the places where it is used. I will make a note to review typing once the services and repositories are reivewed

Reply:
I updated it to use DatasetBase and checked that it works fine